src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -420,15 +438,21 @@ class DetailsCard extends React.Component {
           break
       }
 
-      if (updates !== this.state.vm) {
+      if (updates !== this.state.vm || initTimezoneUpdates !== initTimezoneInfo) {
```

Comment:
The check `initTimezoneUpdates !== initTimezoneInfo` has no effect since we compare the same object (which is a native mutable JS object). IMHO the code works because every update to `initTimezoneUpdates` also re-creates immutable `updates` object. However this will fail if `updates.setIn()` will be used with the same value as currently stored.